### PR TITLE
Create an experiment with all sub-agents active and idle

### PR DESCRIPTION
**Pull request by GeorgeHahn**
### What does this PR do?

Add an SMP experiment with all sub-agents active and idle. This experiment will be used in [Agent Quality Gates](https://datadoghq.atlassian.net/wiki/spaces/SMP/pages/3853091678/Agent+Quality+Gates).

### Motivation

See the Agent Quality Gates doc.

### Additional Notes

This experiment does not currently have any bounds check defined for it. We'd like to let it run for a couple of days to determine an appropriate starting point.

### Describe how to test/QA your changes

This will show up in the Regression Detector PR report.

### CODE REVIEW

**Comment by GeorgeHahn** [comment]
I did a [short analysis](https://app.datadoghq.com/notebook/9388891/checking-agent-idle-all-features-experiment) on the data from this experiment; it looks good to me.

File: test/regression/cases/idle_all_features/lading/lading.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+generator: []
+
+blackhole:
+  - http:
+      binding_addr: "127.0.0.1:9091"
+  - http:
+      binding_addr: "127.0.0.1:9092"
+
+target_metrics:
+  - prometheus:
```
Comment (scottopell) [inline test/regression/cases/idle_all_features/lading/lading.yaml:10]: @cmetz100 please add other target metrics here

**Comment by GeorgeHahn** [comment]
I ran this in a quality gates run and received the following:

---


# Regression Detector Results

Run ID: a43a74c8-61b9-44e4-9f2e-dedba9843f9d [Metrics dashboard](https://app.datadoghq.com/dashboard/ykh-ua8-vcu/SMP-Regression-Detector-Metrics?fromUser=true&refresh_mode=paused&tpl_var_run-id%5B0%5D=a43a74c8-61b9-44e4-9f2e-dedba9843f9d&view=spans&from_ts=1727293998000&to_ts=1727294598000&live=false) [Target profiles](https://app.datadoghq.com/profiling/explorer?query=env%3Asingle-machine-performance%20service%3Adatadog-agent%20job_id%3Aa43a74c8-61b9-44e4-9f2e-dedba9843f9d&agg_m=count&agg_m_source=base&agg_t=count&fromUser=false&viz=stream&start=1727286798000&end=1727298198000&paused=true)

Baseline: 7.54.1
Comparison: nightly-main-3ca96b63-py3

Performance changes are noted in the **perf** column of each table:

* ✅ = significantly better comparison variant performance
* ❌ = significantly worse comparison variant performance
* ➖ = no significant change in performance

## No significant changes in experiment optimization goals

**Confidence level:** 90.00%
**Effect size tolerance:** |Δ mean %| ≥ 5.00%

There were no significant changes in experiment optimization goals at this confidence level and effect size tolerance.


<details>
<summary><h2>
Fine details of change detection per experiment
</h2></summary>

| perf | experiment        | goal               | Δ mean % | Δ mean % CI    | trials | links                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
|------|-------------------|--------------------|----------|----------------|--------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ➖   | idle_all_features | memory utilization | -0.06    | [-0.11, -0.01] | 1      | [Logs](https://app.datadoghq.com/logs?query=experiment%3Aidle_all_features%20run_id%3Aa43a74c8-61b9-44e4-9f2e-dedba9843f9d&agg_m=count&agg_m_source=base&agg_q=%40span.url&agg_q_source=base&agg_t=count&fromUser=true&index=single-machine-performance-target-logs&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=time%2Cdesc&top_n=100&top_o=top&viz=stream&x_missing=true&from_ts=1727286798000&to_ts=1727298198000&live=false)                                                                                                                                                                                                                                                                                                                      |
| ➖   | idle              | memory utilization | -2.04    | [-2.07, -2.02] | 1      | [Logs](https://app.datadoghq.com/logs?query=experiment%3Aidle%20run_id%3Aa43a74c8-61b9-44e4-9f2e-dedba9843f9d&agg_m=count&agg_m_source=base&agg_q=%40span.url&agg_q_source=base&agg_t=count&fromUser=true&index=single-machine-performance-target-logs&messageDisplay=inline&refresh_mode=paused&storage=hot&stream_sort=time%2Cdesc&top_n=100&top_o=top&viz=stream&x_missing=true&from_ts=1727286798000&to_ts=1727298198000&live=false) [Bounds Checks Dashboard](https://app.datadoghq.com/dashboard/vz3-jd5-bdi?fromUser=true&refresh_mode=paused&tpl_var_experiment%5B0%5D=idle&tpl_var_job_id%5B0%5D=a43a74c8-61b9-44e4-9f2e-dedba9843f9d&tpl_var_run-id%5B0%5D=a43a74c8-61b9-44e4-9f2e-dedba9843f9d&view=spans&from_ts=1727293998000&to_ts=1727294598000&live=false) |
</details>

## Bounds Checks

| perf | experiment | bounds_check_name | replicates_passed |
|------|------------|-------------------|-------------------|
| ✅   | idle       | memory_usage      | 30/30             |

<details>
<summary><h2>
Explanation
</h2></summary>

A regression test is an A/B test of target performance in a repeatable rig, where "performance" is measured as "comparison variant minus baseline variant" for an optimization goal (e.g., ingress throughput). Due to intrinsic variability in measuring that goal, we can only estimate its mean value for each experiment; we report uncertainty in that value as a 90.00% confidence interval denoted "Δ mean % CI".

For each experiment, we decide whether a change in performance is a "regression" -- a change worth investigating further -- if all of the following criteria are true:

1. Its estimated |Δ mean %| ≥ 5.00%, indicating the change is big enough to merit a closer look.

2. Its 90.00% confidence interval "Δ mean % CI" does not contain zero, indicating that *if our statistical model is accurate*, there is at least a 90.00% chance there is a difference in performance between baseline and comparison variants.

3. Its configuration does not mark it "erratic".
</details>

**Review by blt — APPROVED** [review]

**Pull request merged**